tests/lean/interactive/run.lean

```diff
@@ -69,6 +80,58 @@ partial def main (args : List String) : IO Unit := do
           for diag in diags do
             IO.eprintln (toJson diag.param)
           requestNo := requestNo + 1
+        | "widgets" =>
+          -- [todo] dedup with RPC
+          if rpcSessionId.isNone then
+            Ipc.writeRequest ⟨requestNo, "$/lean/rpc/connect",  RpcConnectParams.mk uri⟩
+            let r ← Ipc.readResponseAs requestNo RpcConnected
+            rpcSessionId := some r.result.sessionId
+            requestNo := requestNo + 1
+          let tdpp : TextDocumentPositionParams := {textDocument := {uri := uri}, position := pos }
+          let ps : RpcCallParams := {
+            textDocument := {uri := uri},
+            position := pos,
+            sessionId := rpcSessionId.get!,
+            method := `Lean.Widget.getWidgets,
+            params := toJson tdpp,
+          }
+          Ipc.writeRequest ⟨requestNo, "$/lean/rpc/call", ps⟩
+          let response ← Ipc.readResponseAs requestNo Lean.Widget.GetWidgetsResponse
+          requestNo := requestNo + 1
+          IO.eprintln (toJson response.result)
+          for w in response.result.widgets do
+            let params : Lean.Widget.GetWidgetSourceParams := { pos := tdpp, hash := w.javascriptHash }
+            let ps : RpcCallParams := {
+              ps with
+              method := `Lean.Widget.getWidgetSource,
+              params := toJson params,
+            }
+            Ipc.writeRequest ⟨requestNo, "$/lean/rpc/call", ps⟩
+            let resp ← Ipc.readResponseAs requestNo Lean.Widget.WidgetSource
+            IO.eprintln (toJson resp.result)
+            requestNo := requestNo + 1
+        | "RPC" =>
```

Comment:
This call does not seem used in any test, unless I am missing something?

Reply:
no it's not I did use it when I was writing the tests but then I switched to widget because it needs to do two rpc calls to get the widget info

Reply:
I can remove if you like

Reply:
removed